src/libs/pulumi-cli.ts

```diff
@@ -43,6 +57,22 @@ export async function downloadCli(range: string): Promise<void> {
 
   core.info(`Configured range: ${range}`);
 
+  // Check for version of Pulumi CLI installed on the runner
+  const runnerVersion = await getVersion();
+
+  if (runnerVersion) {
+    // Check if runner version matches
+    if (semver.satisfies(runnerVersion, range)) {
+      // If runner version matches, skip downloading CLI by exiting the function
+      core.info(`Pulumi version ${runnerVersion} is already installed on this machine. Skipping download`);
+      return;
+    } else {
+      core.info(`Pulumi ${runnerVersion} does not satisfy the desired version ${range}. Proceeding to download`);
+    }
+  } else {
+    core.info('Pulumi is not detected in the PATH. Proceeding to download');
```

Comment:
Maybe we should use debug here since we're already sending so many info logs?

```suggestion
    core.debug('Pulumi is not detected in the PATH. Proceeding to download');
```

Reply:
Seems like if this last log message is switched to debug the other two should follow suit since they're all part of the same set of conditions (the outer if and the inner if). Only one of the three will be written out so I guess the question is _are these messages helpful in general or should they only be used for troubleshooting?_